src/features/pythonApi.ts

```diff
@@ -171,32 +171,84 @@ class PythonEnvironmentApiImpl implements PythonEnvironmentApi {
             await Promise.all(this.envManagers.managers.map((manager) => manager.refresh(scope)));
             return Promise.resolve();
         }
-        const manager = this.envManagers.getEnvironmentManager(scope);
+        let currentScope: GetEnvironmentScope = scope;
+        if (scope instanceof Uri && scope.scheme === 'vscode-notebook-cell') {
+            currentScope = Uri.from({
+                scheme: 'vscode-notebook',
+                path: scope.path,
+                authority: scope.authority,
+            });
+        }
+        const manager = this.envManagers.getEnvironmentManager(currentScope);
         if (!manager) {
-            return Promise.reject(new Error(`No environment manager found for: ${scope.fsPath}`));
+            return Promise.reject(new Error(`No environment manager found for: ${currentScope.fsPath}`));
         }
-        return manager.refresh(scope);
+        return manager.refresh(currentScope);
     }
     async getEnvironments(scope: GetEnvironmentsScope): Promise<PythonEnvironment[]> {
-        if (scope === 'all' || scope === 'global') {
-            const promises = this.envManagers.managers.map((manager) => manager.getEnvironments(scope));
+        let currentScope: GetEnvironmentsScope = scope;
+        if (currentScope instanceof Uri && currentScope.scheme === 'vscode-notebook-cell') {
+            currentScope = Uri.from({
+                scheme: 'vscode-notebook',
+                path: currentScope.path,
+                authority: currentScope.authority,
+            });
+        }
+        if (currentScope === 'all' || currentScope === 'global') {
+            const promises = this.envManagers.managers.map((manager) => manager.getEnvironments(currentScope));
             const items = await Promise.all(promises);
             return items.flat();
         }
-        const manager = this.envManagers.getEnvironmentManager(scope);
+        const manager = this.envManagers.getEnvironmentManager(currentScope);
         if (!manager) {
             return [];
         }
 
-        const items = await manager.getEnvironments(scope);
+        const items = await manager.getEnvironments(currentScope);
         return items;
     }
     onDidChangeEnvironments: Event<DidChangeEnvironmentsEventArgs> = this._onDidChangeEnvironments.event;
     setEnvironment(scope: SetEnvironmentScope, environment?: PythonEnvironment): Promise<void> {
-        return this.envManagers.setEnvironment(scope, environment);
+        let currentScope: SetEnvironmentScope = scope;
+        if (scope instanceof Uri && scope.scheme === 'vscode-notebook-cell') {
```

Comment:
This seems to be duplicated a lot. We could consolidate this.

Reply:
addressed